src/analysis/retail/shaman/enhancement/modules/normalizers/MaelstromWeaponBuffNormalizer.tsx

```diff
@@ -0,0 +1,55 @@
+import EventsNormalizer from 'parser/core/EventsNormalizer';
+import { AnyEvent, EventType, HasAbility } from 'parser/core/Events';
+import SPELLS from 'common/SPELLS/shaman';
+import { Options } from 'parser/core/Analyzer';
+
+/**
+ * This normalizer removes the duplicate 'refreshbuff' events that immediately follow
+ * every 'applybuff' and 'applybuffstack' event, leaving 'refreshbuff' events exclusively for
+ * gains at cap
+ */
+class MaelstromWeaponBuffNormalizer extends EventsNormalizer {
+  constructor(options: Options) {
+    super(options);
+    this.priority = -100;
+  }
+
+  normalize(events: AnyEvent[]): AnyEvent[] {
+    const fixedEvents: AnyEvent[] = [];
+    const skipEvents: number[] = [];
+    events.forEach((event: AnyEvent, idx: number) => {
+      if (HasAbility(event) && event.ability.guid === SPELLS.MAELSTROM_WEAPON_BUFF.id) {
+        if (event.type === EventType.ApplyBuff || event.type === EventType.ApplyBuffStack) {
+          fixedEvents.push(event);
+          // look for a refresh event following this one
+          for (let forwardIndex = idx; forwardIndex < events.length; forwardIndex += 1) {
+            const forwardEvent = events[forwardIndex];
+            if (forwardEvent.timestamp - event.timestamp > 10) {
+              break;
+            }
+            if (
+              forwardEvent.type === EventType.RefreshBuff &&
+              forwardEvent.ability.guid === SPELLS.MAELSTROM_WEAPON_BUFF.id
+            ) {
+              skipEvents.push(forwardIndex);
+              break;
+            }
+          }
+        } else if (event.type === EventType.RefreshBuff && !skipEvents.includes(idx)) {
```

Comment:
`skipEvents` will probably be pretty large, right? maybe want to use a `Set` here instead of an array

Reply:
Good idea. I've not used Sets in JS/TS but upon looking into them this is a much better idea.

In terms of size, the array is unlikely to contain more than 500 elements for a raid although many more could be in mythic+ logs.